### PR TITLE
Change `@transform` on a grouped dataframe to match DataFrames.transform

### LINKED ISSUE #176: Nested transform is broken on master branch
Given this data frame:
```
julia> d = DataFrame(rand(2,3))
2×3 DataFrame
│ Row │ x1        │ x2       │ x3       │
│     │ Float64   │ Float64  │ Float64  │
├─────┼───────────┼──────────┼──────────┤
│ 1   │ 0.876809  │ 0.385893 │ 0.111728 │
│ 2   │ 0.0609952 │ 0.15999  │ 0.100248 │
```

A plain transform works fine:
```
julia> @transform(d, y1 = :x3 .* 2)
2×4 DataFrame
│ Row │ x1        │ x2       │ x3       │ y1       │
│     │ Float64   │ Float64  │ Float64  │ Float64  │
├─────┼───────────┼──────────┼──────────┼──────────┤
│ 1   │ 0.876809  │ 0.385893 │ 0.111728 │ 0.223456 │
│ 2   │ 0.0609952 │ 0.15999  │ 0.100248 │ 0.200496 │
```

But if I nest the result of the previous transform with another one, it doesn't work:
```
julia> @transform(@transform(d, y1 = :x3 .* 2), y2 = :y1 .* 2)
2×4 DataFrame
│ Row │ x1        │ x2       │ x3       │ y1       │
│     │ Float64   │ Float64  │ Float64  │ Float64  │
├─────┼───────────┼──────────┼──────────┼──────────┤
│ 1   │ 0.876809  │ 0.385893 │ 0.111728 │ 0.223456 │
│ 2   │ 0.0609952 │ 0.15999  │ 0.100248 │ 0.200496 │
```

If I assign it to a temp variable then it works properly. So it's probably an issue with the macro.
```
julia> d_temp = @transform(d, y1 = :x3 .* 2)
2×4 DataFrame
│ Row │ x1        │ x2       │ x3       │ y1       │
│     │ Float64   │ Float64  │ Float64  │ Float64  │
├─────┼───────────┼──────────┼──────────┼──────────┤
│ 1   │ 0.876809  │ 0.385893 │ 0.111728 │ 0.223456 │
│ 2   │ 0.0609952 │ 0.15999  │ 0.100248 │ 0.200496 │

julia> @transform(d_temp, y2 = :y1 .* 2)
2×5 DataFrame
│ Row │ x1        │ x2       │ x3       │ y1       │ y2       │
│     │ Float64   │ Float64  │ Float64  │ Float64  │ Float64  │
├─────┼───────────┼──────────┼──────────┼──────────┼──────────┤
│ 1   │ 0.876809  │ 0.385893 │ 0.111728 │ 0.223456 │ 0.446911 │
│ 2   │ 0.0609952 │ 0.15999  │ 0.100248 │ 0.200496 │ 0.400992 │
```

My versions:
```
(DataFramesMetaTest) pkg> st
Status `~/Julia/DataFramesMetaTest/Project.toml`
  [a93c6f00] DataFrames v0.21.7
  [1313f7d8] DataFramesMeta v0.5.1 `https://github.com/JuliaData/DataFramesMeta.jl#master`

julia> versioninfo()
Julia Version 1.5.1
Commit 697e782ab8 (2020-08-25 20:08 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.5.0)
  CPU: Intel(R) Core(TM) i5-4258U CPU @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, haswell)
Environment:
  JULIA_NUM_THREADS = 4
```

### PR DESCRIPTION
**Pull request by pdeffebach**
also add tests for `@select`, which already worked after #163 

Closes #176 and superscedes #178

### CODE REVIEW

**Comment by pdeffebach** [comment]
Not sure why the travis build is stalled. But this passes tests locally. 

I will merge after 1 approval since it's a small change that has a consensus. 

I'm not adding docs yet because the docs need a big re-write and the docus don't currently mention this particular feature.

**Review by bkamins — APPROVED** [review]

**Comment by bkamins** [comment]
CI was stalled, because yesterday we had contributors who pushed many small commits to JuliaData - and this is a highly problematic situation (especially if these are commits to DataFrames.jl, whose CI lasts ~1h per commit).

File: test/grouping.jl [hunk 1]
```diff
@@ -219,46 +219,57 @@ end
...
+	g = groupby(d, :x)
 
-	@test (@transform(g, y = :n .- median(:n)))[1,:y] == -5.0
+	@test (@transform(g, y = :n .- median(:n)))[1,:y] == -2.0
 
 	d = DataFrame(a = [1,1,1,2,2,3,3,1],
 	              b = Any[1,2,3,missing,missing,6.0,5.0,4],
 	              c = CategoricalArray([1,2,3,1,2,3,1,2]))
-      g = groupby(d, :a)
+    g = groupby(d, :a)
```
Comment (nalimilan) [inline test/grouping.jl:229]: Indentation seems to be incorrect here and in a few other places.

File: News.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# DataFramesMeta v0.6 Release Notes
```
Comment (nalimilan) [inline News.md:1]: Isn't it more common to call this file NEWS.md in uppercase?

File: News.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+* The order of rows after a `@transform(gd::GroupedDataFrame, args...)` now matches the 
+order of rows returned after `DataFrames.transform(gd::GroupedDataFrame, args...)`. 
+`@select` now supports `GroupedDataFrame` ([#180])
```
Comment (nalimilan) [inline News.md:5]:
```suggestion
  order of rows returned after `DataFrames.transform(gd::GroupedDataFrame, args...)`. 
  `@select` now supports `GroupedDataFrame` ([#180])
```
(I think)

File: test/grouping.jl (hunk 1 above)
Comment (pdeffebach) [reply test/grouping.jl:229]: indendation looks correct locally. I think github is displaying things weirdly.

File: News.md (hunk 3 above)
Comment (pdeffebach) [reply News.md:5]: Added a new bullet for `@select`, indentation as requested.

File: News.md (hunk 2 above)
Comment (pdeffebach) [reply News.md:1]: added.

File: test/grouping.jl (hunk 1 above)
Comment (nalimilan) [reply test/grouping.jl:229]: Looks like the rest of the file uses tabs. Better use tabs too, and fix the whole file later.

Comment (pdeffebach) [reply test/grouping.jl]: Should be fixed now. Thanks for finding that.

**Review by nalimilan — APPROVED** [review]

**Pull request merged**